### ISSUE TITLE
Conflicts between `avoid_as` and `strong_mode: implicit-casts: false`

### ISSUE THREAD

**Issue by alextekartik** [dart-lang/linter#1400]
Using flutter master 1.1.10-pre.136, dart 2.1.1

```
Flutter 1.1.10-pre.136 • channel master • https://github.com/flutter/flutter.git Framework • revision 2eee7b7e94 (10 days ago) • 2019-01-20 20:28:39 -0500 Engine • revision 5a8b0f3dba Tools • Dart 2.1.1 (build 2.1.1-dev.2.0 2e5453ddb4)
```
Consider the following analysis_options.yaml

```yaml
analyzer:
  strong-mode:
    implicit-casts: false

linter:
  rules:
  - avoid_as
```
Consider the following code

```dart
Map<String, dynamic> map = {};

// hint: Avoid using as
final text1 = map['value'] as String;
// error: A value of type 'dynamic' can't be assigned to a variable of type 'String'
final String text2 = map['value'];
```

I could not find a solution without removing one of these 2 rules. What would be the solution (i.e. read an expected String from a map) and is there any planned modification on these?

Thanks!

**Comment by bwilkerson** [comment]
There really isn't a conflict here. Both are related to run-time casts. The `strong_mode: implicit-casts: false` option allows you to prohibit implicit casts while `avoid_as` allows you to prohibit explicit casts.

If you really want to prohibit both explicit and implicit run-time casts, I can think of two options.

1. Explicitly test the type by doing something like this:
```dart
final value = map['value'];
if (value is String) {
  final text = value;
  // Use the text in some way.
} else {
  // Do whatever is appropriate when the value is not a String.
}
```

2. Remove the need for the test my defining `map` to be a `Map<String, String>`. Of course, that isn't always an option, but it's worth considering when it is possible.

**Comment by alextekartik** [comment]
Thanks. I think I will remove `avoid_as` for now (I read also bool, int and other types), as `implicit-casts: false` helps on finding potential runtime issue at compile time
